src/deepsparse/clip/pipelines.py

```diff
@@ -0,0 +1,249 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Tuple, Type
+
+import numpy as np
+import onnx
+from numpy import linalg as la
+from PIL import Image
+from torchvision import transforms
+from torchvision.transforms import InterpolationMode
+
+from deepsparse.clip.constants import CLIP_RGB_MEANS, CLIP_RGB_STDS
+from deepsparse.clip.schemas import (
+    CLIPTextInput,
+    CLIPTextOutput,
+    CLIPVisualInput,
+    CLIPVisualOutput,
+    CLIPZeroShotInput,
+    CLIPZeroShotOutput,
+)
+from deepsparse.pipeline import BasePipeline, Pipeline
+from deepsparse.utils import model_to_path
+from open_clip.tokenizer import tokenize
+from scipy.special import softmax
+
+
+__all__ = ["CLIPVisualPipeline", "CLIPTextPipeline", "CLIPZeroShotPipeline"]
```

Comment:
potential refactor - instead of pipelines and schemas files, might be good to have "visual_pipeline" standalone file that contains the specific pipeline and its schemas